content/academy/data_collection_basics/api_scraping.md

```diff
@@ -122,9 +127,61 @@ const HEADERS = {
+
+With `got-scraping`, generating request-specific headers can be done right within a request with `headerGeneratorOptions`. Specific headers can also be set with the "`headers`" option:
 
-For our SoundCloud example, testing the endpoint from the previous section in a tool like <a href="https://www.postman.com/">Postman</a> works perfectly, and returns the data we want; however, when the "`client_id`" parameter is removed, we receive a `401 Unauthorized` error. Luckily, the Client ID is the same for every user, which means that it is not tied to a session or an IP address (this is based on our own observations and tests). The big downfall is that the token being used by SoundCloud changes every few weeks, so it shouldn't be hardcoded. **A very strange case indeed**.
+```JavaScript
+const response = await gotScraping({
+    url: 'https://example.com',
+    headerGeneratorOptions:{
+        browsers: [
+            {
+                name: 'chrome',
+                minVersion: 87,
+                maxVersion: 89
+            }
+        ],
+        devices: ['desktop'],
+        locales: ['de-DE', 'en-US'],
+        operatingSystems: ['windows', 'linux'],
+    },
+    headers: {
+        'some-header': 'Hello, Academy!'
+    }
+})
+```
+
+For our SoundCloud example, testing the endpoint from the previous section in a tool like [Postman](https://www.postman.com/) works perfectly, and returns the data we want; however, when the "`client_id`" parameter is removed, we receive a `401 Unauthorized` error. Luckily, the Client ID is the same for every user, which means that it is not tied to a session or an IP address (this is based on our own observations and tests). The big downfall is that the token being used by SoundCloud changes every few weeks, so it shouldn't be hardcoded. This case is actually quite common, and is not only seen with SoundCloud.
+
+Ideally, this `client_id` should be scraped dynamically, especially since it changes frequently, but unfortunately, the token cannot be found anywhere on SoundCloud's pages. We already know that it's available within the parameters of certain requests though, and luckily, [Puppeteer](https://github.com/puppeteer/puppeteer) offers a simple way to analyze each response when on a page. It's a bit like using browser DevTools, which you are already familiar with by now, but programatically instead.
+
+Here is a way you could dynamically scrape the `client_id` using Puppeteer:
+
+```JavaScript
+const puppeteer = require('puppeteer'); // import the puppeteer module
+
+const scrapeClientId = async () => {
+    const browser = await puppeteer.launch({ headless: false });
+    const page = await browser.newPage();
+
+    let clientId = null; // initialize a variable that will eventually hold the client_id
```

Comment:
I would place comments on top of the code, instead of next to it. On smaller screens, they won't wrap that much.